sourcegraph/cody
# Autocomplete: Add section history to graph context

This PR starts to keep track of all the code sections that a user is navigated into (i.e if they move the cursor into a section) and uses sections they have previously visited as strong references for context.

The idea here is if you look at section A (say, the `bubbleSort()` function) and then move to somewhere else (e.g. a new test file), chances are that what you looked at previously is relevant to the new section too (e.g. you might want to write a test for `bubbleSort()`).

## Test plan

- Enable graph context
- Create a file with a `bubbleSort` implementation
- Create a new file and start a `describe()` function
- Notice that we have context of the previous function now:

<img width="739" alt="Screenshot 2023-09-11 at 17 45 38" src="https://github.com/sourcegraph/cody/assets/458591/92472709-40a5-43c6-b41f-6591e961120d">
